File: 天梯赛/java/L1-007. 念数字.java
import java.util.Scanner;

public class Main {
	public static final String[] chinese= {"ling","yi","er","san","si","wu","liu","qi","ba","jiu"};
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		String n=in.nextLine();
		in.close();
		boolean first=true;
		if(n.charAt(0)=='-') {
			System.out.print("fu ");
			first=false;
		}
		for(int i=(first)?0:1;i<n.length();++i) {
			System.out.print(chinese[n.charAt(i)-'0']);
			if(i!=n.length()-1)System.out.print(" ");
		}
	}	
}

File: 天梯赛/L1-007. 念数字.cpp
#include<iostream>
#include<cstring>
#include<vector>
#include<string>
#include<cstring>
#include<queue>
#include<set>
#include<stack>
#include<map>
#include<cmath>
#include<cstdio>
#include<algorithm>
using namespace std;
#define lowbit(i) ((i)&(-i))
typedef long long ll;
const int maxn=1e5+10;
const int inf=0x3f3f3f3f;
map<char,string> m;
int main(){
	ios::sync_with_stdio(false);
	string s;
	cin>>s;
	m[0]="ling";m[1]="yi";m[2]="er";
	m[3]="san";m[4]="si";m[5]="wu";
	m[6]="liu";m[7]="qi";m[8]="ba";
	m[9]="jiu";
	if(s[0]=='-'){
		cout<<"fu ";
		s.erase(0,1);
	}
	for(int i=0;i<s.length();++i){
		if(i)cout<<" ";
		cout<<m[s[i]-'0'];
	}
	return 0;
}
